samplers/jaegerremote/sampler_remote_options.go

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+// Copyright (c) 2021 The Jaeger Authors.
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerremote
+
+import (
+	"time"
+
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type config struct {
+	sampler                 trace.Sampler
+	samplingServerURL       string
+	samplingRefreshInterval time.Duration
+	samplingFetcher         samplingStrategyFetcher
+	samplingParser          samplingStrategyParser
+	updaters                []samplerUpdater
+	posParams               perOperationSamplerParams
+}
+
+// newConfig returns an appropriately configured config.
+func newConfig(options ...Option) config {
+	c := config{
+		sampler:                 newProbabilisticSampler(0.001),
+		samplingServerURL:       defaultSamplingServerURL,
+		samplingRefreshInterval: defaultSamplingRefreshInterval,
+		samplingFetcher:         newHTTPSamplingStrategyFetcher(defaultSamplingServerURL),
+		samplingParser:          new(samplingStrategyParserImpl),
+		updaters: []samplerUpdater{
+			&perOperationSamplerUpdater{
+				MaxOperations:            defaultSamplingMaxOperations,
+				OperationNameLateBinding: defaultSamplingOperationNameLateBinding,
+			},
+			new(probabilisticSamplerUpdater),
+			new(rateLimitingSamplerUpdater),
+		},
+	}
+	for _, option := range options {
+		option.apply(&c)
+	}
+
+	return c
```

Comment:
```suggestion
		updaters: []samplerUpdater{
			new(probabilisticSamplerUpdater),
			new(rateLimitingSamplerUpdater),
		},
		posParams: perOperationSamplerParams{
			MaxOperations: defaultSamplingMaxOperations,
			OperationNameLateBinding: defaultSamplingOperationNameLateBinding,
		},
	}
	for _, option := range options {
		option.apply(&c)
	}

	c.updaters = append(c.updaters,
		&perOperationSamplerUpdater{
			MaxOperations:            c.posParams.MaxOperations,
			OperationNameLateBinding: c.posParams.OperationNameLateBinding,
		})
			
	return c
```

Reply:
This would change the order in which updaters are invoked. I would recommend minimizing superficial changes in this PR as it was supposed to be mostly a straight copy of the existing (prod quality) code from Jaeger SDK.

Reply:
Then append the existing updaters to the new one:

```go
c.updaters = append([]samplerUpdater(&perOperartionSamplerUpdater{...}), c.updaters...)
```

It's not a hard problem to solve, but the configuration for it doesn't work as is, so it's not a superficial change.

Reply:
done